cmd/notary/integration_test.go

```diff
@@ -81,6 +81,43 @@ func setupServer() *httptest.Server {
 	return httptest.NewServer(setupServerHandler(storage.NewMemStorage()))
 }
 
+// Initializes a repo with existing key
+func TestInitWithMultipleRootKeys(t *testing.T) {
+	// -- setup --
+	setUp(t)
+
+	tempDir := tempDirWithConfig(t, "{}")
+	defer os.RemoveAll(tempDir)
+
+	server := setupServer()
+	defer server.Close()
+
+	tempFile, err := ioutil.TempFile("", "targetfile")
+	require.NoError(t, err)
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	// -- tests --
+
+	// generate root key produces a single root key and no other keys
+	_, err = runCommand(t, tempDir, "key", "generate", data.ECDSAKey)
+	require.NoError(t, err)
+	rootIDs, _ := assertNumKeys(t, tempDir, 1, 0, true)
+	var rootFiles []string
+	for _, rID := range rootIDs {
+		rootFiles = append(rootFiles, filepath.Join(
+			tempDir, "private", "root_keys", rID+".key"))
+	}
+
+	// init repo
+	_, err = runCommand(t, tempDir, "-s", server.URL, "init", "gun", "--rootkey", rootFiles[0])
+	require.NoError(t, err)
+
+	// publish repo
+	_, err = runCommand(t, tempDir, "-s", server.URL, "publish", "gun")
+	require.NoError(t, err)
```

Comment:
Sorry @cyli GitHub caching caused me to miss your comment. 

The integration tests seem to only test the "happy path" - not saying that's all they should test, just noting that it's in keeping with the current style.

IIRC the cases you listed are tested at the api levels, because the addition in this PR uses all the same code paths for key loading. I will double check to make sure.